qutebrowser/mainwindow/prompt.py

```diff
@@ -633,6 +635,42 @@ def __init__(self, question, parent=None):
         self._to_complete = ''
         self._root_index = QModelIndex()
 
+        self._history = cmdhistory.History(parent=self)
+        try:
+            fprompt_history = objreg.get('fprompt-history')
+            self._history.history = fprompt_history.data
+            self._history.changed.connect(fprompt_history.changed)
+            self._shortcut_next = QShortcut(QKeySequence("Ctrl+P"), self)
+            self._shortcut_next.activated.connect(self.history_next)
+            self._shortcut_prev = QShortcut(QKeySequence("Ctrl+N"), self)
```

Comment:
Please expose those as qutebrowser commands and add normal keybindings instead of hardcoded shortcuts.

Reply:
Ok will do that. What file do you recommend I register them in because I don't know enough about the project to know where. Should I put it in the FilenamePrompt class?

Reply:
They will need to be on the `PromptContainer` to work properly. Please take a look at how the other `:prompt-*` commands are implemented earlier in this file.